homeassistant/components/synology_dsm/camera.py

```diff
@@ -31,20 +32,25 @@
 
 
 async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable
 ) -> None:
     """Set up the Synology NAS cameras."""
 
     data = hass.data[DOMAIN][entry.unique_id]
-    api = data[SYNO_API]
+    api: SynoApi = data[SYNO_API]
 
     if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
         return
 
     # initial data fetch
-    coordinator = data[COORDINATOR_CAMERAS]
+    coordinator: DataUpdateCoordinator[dict[str, dict[str, SynoCamera]]] = data[
+        COORDINATOR_CAMERAS
+    ]
```

Comment:
Let's cleanup this a bit. `async_config_entry_first_refresh` is recommended now. Replace
```python
await coordinator.async_refresh()
if coordinator.data is None:
  raise ConfigEntryNotReady
```
with
```python
await coordinator.async_config_entry_first_refresh()
assert coordinator.data is not None
```